### ISSUE TITLE
blacklist not working on fedora

### ISSUE THREAD

**Issue by mariusagur** [aigilea/hp_spectre_x360_14_eu0xxx#4]
Hi! I've tried the blacklist and it works for ubuntu, arch and opensuse, but not my preferred distro fedora 
Tried different spins and everything from fedora 37 to rawhide.
I get instant hard lock after initrd loads (added echo statements to grub menu), caps lock light doesn't respond, ctrl+alt+del not responding, etc.
Tried both modprobe.blacklist and module_blacklist, also tried noapic, acpi=off, loglevel=7, blacklisted intel_plss in addition to pci module, tried acpi_osi linux and windows 2017.

Any idea what could be causing this specifically for fedora?

**Comment by UnconventionalMindset** [comment]
I believe I saw issues for blissOs and android x86 using older kernels. Maybe check if the kernel version that Fedora uses is recent enough? Not sure tho.

**Comment by mariusagur** [comment]
> I believe I saw issues for blissOs and android x86 using older kernels. Maybe check if the kernel version that Fedora uses is recent enough? Not sure tho.

No, fc39 har 6.5 kernel, FC40 has 6.8 kernel, rawhide has 6.9rc3 kernel.

**Comment by jkaplavka** [comment]
Hi, I don't know why blacklisting doesn't work in fedora, but I was able to install Fedora 40 beta by modifying installation iso. I don't remember exactly all commands, but maybe it will help you.

- boot live Ubuntu 24.04 with `modprobe.blacklist=intel_lpss_pci`
- patch and compile dsdt.aml file (readme step 6 and 7)
- modify fedora iso, copy dsdt.aml to /boot/ to iso with xorriso tool

```sh
xorriso -indev Fedora-Workstation-Live-x86_64-40_Beta-1.2.iso -outdev modified.iso -compliance no_emul_toc -map dsdt.aml /boot/dsdt.aml -boot_image any replay
```

- now you should be able to boot `modified.iso`, at grub press `e` and add `acpi /boot/dsdt.aml`
- install fedora but don't reboot, you have to change grub config to apply dsdt.aml
- chroot into installed fedora, by following steps in https://gist.github.com/Tamal/73e65bfb0e883e438310c5fe81c5de14 until line 24
- copy dsdt.aml to `/boot/dsdt.aml`
- create file `/etc/grub.d/09_hp_acpi`

```
#!/usr/bin/sh -e

# Uncomment to load custom ACPI table
GRUB_CUSTOM_ACPI="/boot/dsdt.aml"

# DON'T MODIFY ANYTHING BELOW THIS LINE!

prefix=/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib

. "$pkgdatadir/grub-mkconfig_lib"

# Load custom ACPI table
if [ x${GRUB_CUSTOM_ACPI} != x ] && [ -f ${GRUB_CUSTOM_ACPI} ] \
        && is_path_readable_by_grub ${GRUB_CUSTOM_ACPI}; then
    echo "Found custom ACPI table: ${GRUB_CUSTOM_ACPI}" >&2
    prepare_grub_to_access_device `${grub_probe} --target=device ${GRUB_CUSTOM_ACPI}` | sed -e "s/^/  /"
    cat << EOF
acpi (\$root)`make_system_path_relative_to_its_root ${GRUB_CUSTOM_ACPI}`
EOF
fi

```

- regenerate grub with `grub2-mkconfig -o /boot/efi/EFI/fedora/grub.cfg` and exit chroot
- now installed fedora should boot

**Comment by mariusagur** [comment]
> Hi, I don't know why blacklisting doesn't work in fedora, but I was able to install Fedora 40 beta by modifying installation iso. I don't remember exactly all commands, but maybe it will help you.
> 
>     * boot live Ubuntu 24.04 with `modprobe.blacklist=intel_lpss_pci`
> 
>     * patch and compile dsdt.aml file (readme step 6 and 7)
> 
>     * modify fedora iso, copy dsdt.aml to /boot/ to iso with xorriso tool
> 
> 
> ```shell
> xorriso -indev Fedora-Workstation-Live-x86_64-40_Beta-1.2.iso -outdev modified.iso -compliance no_emul_toc -map dsdt.aml /boot/dsdt.aml -boot_image any replay
> ```
> 
>     * now you should be able to boot `modified.iso`, at grub press `e` and add `acpi /boot/dsdt.aml`
> 
>     * install fedora but don't reboot, you have to change grub config to apply dsdt.aml
> 
>     * chroot into installed fedora, by following steps in https://gist.github.com/Tamal/73e65bfb0e883e438310c5fe81c5de14 until line 24
> 
>     * copy dsdt.aml to `/boot/dsdt.aml`
> 
>     * create file `/etc/grub.d/09_hp_acpi`
> 
> 
> ```
> #!/usr/bin/sh -e
> 
> # Uncomment to load custom ACPI table
> GRUB_CUSTOM_ACPI="/boot/dsdt.aml"
> 
> # DON'T MODIFY ANYTHING BELOW THIS LINE!
> 
> prefix=/usr
> exec_prefix=${prefix}
> libdir=${exec_prefix}/lib
> 
> . "$pkgdatadir/grub-mkconfig_lib"
> 
> # Load custom ACPI table
> if [ x${GRUB_CUSTOM_ACPI} != x ] && [ -f ${GRUB_CUSTOM_ACPI} ] \
>         && is_path_readable_by_grub ${GRUB_CUSTOM_ACPI}; then
>     echo "Found custom ACPI table: ${GRUB_CUSTOM_ACPI}" >&2
>     prepare_grub_to_access_device `${grub_probe} --target=device ${GRUB_CUSTOM_ACPI}` | sed -e "s/^/  /"
>     cat << EOF
> acpi (\$root)`make_system_path_relative_to_its_root ${GRUB_CUSTOM_ACPI}`
> EOF
> fi
> ```
> 
>     * regenerate grub with `grub2-mkconfig -o /boot/efi/EFI/fedora/grub.cfg` and exit chroot
> 
>     * now installed fedora should boot

Injecting dsdt.aml to iso worked perfectly!
The acpi script did not work for me, though, it says it finds file but will not work on reboot. I changed it to just do this:
acpi (\$root)\dsdt.aml

**Comment by christian-bendiksen** [comment]
> Hi, I don't know why blacklisting doesn't work in fedora, but I was able to install Fedora 40 beta by modifying installation iso. I don't remember exactly all commands, but maybe it will help you.
> 
>     * boot live Ubuntu 24.04 with `modprobe.blacklist=intel_lpss_pci`
> 
>     * patch and compile dsdt.aml file (readme step 6 and 7)
> 
>     * modify fedora iso, copy dsdt.aml to /boot/ to iso with xorriso tool
> 
> 
> ```shell
> xorriso -indev Fedora-Workstation-Live-x86_64-40_Beta-1.2.iso -outdev modified.iso -compliance no_emul_toc -map dsdt.aml /boot/dsdt.aml -boot_image any replay
> ```
> 
>     * now you should be able to boot `modified.iso`, at grub press `e` and add `acpi /boot/dsdt.aml`
> 
>     * install fedora but don't reboot, you have to change grub config to apply dsdt.aml
> 
>     * chroot into installed fedora, by following steps in https://gist.github.com/Tamal/73e65bfb0e883e438310c5fe81c5de14 until line 24
> 
>     * copy dsdt.aml to `/boot/dsdt.aml`
> 
>     * create file `/etc/grub.d/09_hp_acpi`
> 
> 
> ```
> #!/usr/bin/sh -e
> 
> # Uncomment to load custom ACPI table
> GRUB_CUSTOM_ACPI="/boot/dsdt.aml"
> 
> # DON'T MODIFY ANYTHING BELOW THIS LINE!
> 
> prefix=/usr
> exec_prefix=${prefix}
> libdir=${exec_prefix}/lib
> 
> . "$pkgdatadir/grub-mkconfig_lib"
> 
> # Load custom ACPI table
> if [ x${GRUB_CUSTOM_ACPI} != x ] && [ -f ${GRUB_CUSTOM_ACPI} ] \
>         && is_path_readable_by_grub ${GRUB_CUSTOM_ACPI}; then
>     echo "Found custom ACPI table: ${GRUB_CUSTOM_ACPI}" >&2
>     prepare_grub_to_access_device `${grub_probe} --target=device ${GRUB_CUSTOM_ACPI}` | sed -e "s/^/  /"
>     cat << EOF
> acpi (\$root)`make_system_path_relative_to_its_root ${GRUB_CUSTOM_ACPI}`
> EOF
> fi
> ```
> 
>     * regenerate grub with `grub2-mkconfig -o /boot/efi/EFI/fedora/grub.cfg` and exit chroot
> 
>     * now installed fedora should boot

Hey! Got it working on Ubuntu but would like to use Fedora as it is my main distro. However, i am to confused by the steps at `chroot into installed fedora, by following steps in https://gist.github.com/Tamal/73e65bfb0e883e438310c5fe81c5de14 until line 24` as i am getting completely different outputs. Not really sure what my root partition is.

What am i looking for? i saw live-rw, live-base etc.

**Comment by christian-bendiksen** [comment]
Had to use `mount subvol=root <partition> /mnt` to get it working. All was booting fine, thanks.

However my display seems to be locked at 60hz on Fedora, anyone else experience this?

**Comment by syfox0** [comment]
So I seem to still have issued trying to boot into a Live USB of Fedora:

- Created multiple LiveUSB of Fedora 40 WS with Etcher, Rufus, and Fedora Media Writer - unable to boot with `modprobe.blacklist=intel_lpss_pci` (as expected, based on the reported issue above).  
- So I booted into a live ISO of Ubuntu (using the same modprobe parameter), and patched and compiled an .aml file. 
- Created a new ISO using xorriso with the .aml file in /boot/ directory (verified by mounting the ISO to check the boot folder)
- Created a LiveUSB from the new ISO
- When I press `e` and add `acpi /boot/dsdt.aml`, nothing happens (same as before - just a blank screen that says "Booting a command list").

And there is another twist
- On a LiveUSB with Ventoy, Fedora 40 WS(the exact same ISO's as above) starts the boot process with `modprobe.blacklist=intel_lpss_pci`. However, it gets stuck at the following error cycle (with the time count increasing):
`(1 of 2) Job dev-mapper-live\x2drw.device/start running ( x min x s /50min)`
`(1 of 2) Job systemd-udev-trigger.service/start running ( x min x s / no limit)`

Not sure what I am doing wrong but any ideas would be appreciated!

**Comment by mariusagur** [comment]
> So I seem to still have issued trying to boot into a Live USB of Fedora:
> 
>     * Created multiple LiveUSB of Fedora 40 WS with Etcher, Rufus, and Fedora Media Writer - unable to boot with `modprobe.blacklist=intel_lpss_pci` (as expected, based on the reported issue above).
> 
>     * So I booted into a live ISO of Ubuntu (using the same modprobe parameter), and patched and compiled an .aml file.
> 
>     * Created a new ISO using xorriso with the .aml file in /boot/ directory (verified by mounting the ISO to check the boot folder)
> 
>     * Created a LiveUSB from the new ISO
> 
>     * When I press `e` and add `acpi /boot/dsdt.aml`, nothing happens (same as before - just a blank screen that says "Booting a command list").
> 
> 
> And there is another twist
> 
>     * On a LiveUSB with Ventoy, Fedora 40 WS(the exact same ISO's as above) starts the boot process with `modprobe.blacklist=intel_lpss_pci`. However, it gets stuck at the following error cycle (with the time count increasing):
>       `(1 of 2) Job dev-mapper-live\x2drw.device/start running ( x min x s /50min)`
>       `(2 of 2) Job systemd-udev-trigger.service/start running ( x min x s / no limit)`
> 
>     * This also happens if I launch the modified ISO from Ventoy using the `acpi /boot/dsdt.aml` parameter in grub
> 
> 
> Not sure what I am doing wrong but any ideas would be appreciated!

You need to do:

acpi (\$root)\dsdt.aml

**Comment by syfox0** [comment]
hi @mariusagur , thanks for the quick response. I think you are saying I should use `acpi ($root)\dsdt.aml` parameter after presseing `e` in grub - right?

I tried adding that when loading the modified ISO, but unfortunately still the same as before. Just to be safe, i also tried using `acpi ($root)/dsdt.aml` incase the backslash was the issue. I am still stuck at a blank screen that says "Booting a command list"

**Comment by syfox0** [comment]
Also, just wanted to ensure I am doing the Step 6 and 7 of the README correctly:
- I downloaded the `hp-spectre-x360-14-eu0xxx-f5a.asl` file from github
- Ran `iasl -tc hp-spectre-x360-14-eu0xxx-f5a.asl` in the same directory as the above ASL file
- It created a file named `hp-spectre-x360-14-eu0xxx-f5a.aml`. I renamed this file to `dsdt.aml`
- Then ran the xorriso command to this dsdt.aml into the /boot/ directory of the Fedora Workstation ISO (fedora 40 release)

I am new to Linux and have enjoyed tinkering with it so far, so just making sure I am doing the above correctly.

Thanks!

**Comment by syfox0** [comment]
hi @jkaplavka , I was just about to post an update after messing around and finally being able to solve it, but in a slightly different way:

- Instead of pressing `e`, I used `c` for the command line and executed  `acpi ($root)/boot/dsdt.aml`
- Pressed `esc` and booted into the LiveUSB without any issues. Installed fedora normally, and followed the chroot guide except i changed the following:
- Line 4 of the chroot guide, I instead did what christian said above: `mount -o subvol=root <partition of linux filesystem> /mnt`
- for 09_hp_acpi script, I created the file using `touch` and `nano` , but the file was not executable so I noticed that the `grub2-mkconfig` wasn't pulling the script into the `grub.cfg`. So i ran `chmod a+x 09_hp_acpi` and re-ran `grub2-mkconfig` and was able to boot into the installed fedora!

I think what I was doing wrong initially was adding `acpi /boot/dsdt.aml` at the end of the `linuxefi` line in grub. Thanks for the image, that clarifies my mistake. I am a total newbie to this and had a lot of fun with the troubleshooting and learning!

FYI - without any updates/patching, the following is working on F40 Workstation release:
- touchpad, keyboard, touchscreen, pen, wifi and sound
